### ISSUE TITLE
Build failure: sagelib

### ISSUE THREAD

**Issue by ErinaInit** [NixOS/nixpkgs#309738]
### Steps To Reproduce

1. build sagelib

### Build log

The core logs:

```
removing build/bdist.linux-x86_64/wheel
Successfully built sagemath_standard-10.3-cp311-cp311-linux_x86_64.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
buildPhase completed in 36 minutes 14 seconds
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for sagemath_standard-10.3-cp311-cp311-linux_x86_64.whl
  - networkx<3.3,>=2.4 not satisfied by version 3.3
```

The full log is very long and I don't consider it very useful in this case......

We may need to change the dependencies or patch this.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.29, NixOS, 24.05 (Uakari), 24.05.20240502.63c3a29`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/p69bcs7ma6ijj8v9xsrg3nq3nn8ryn95-source`
```

**Comment by ErinaInit** [comment]
This might have been fixed by commit #310665. Let me test.

**Comment by ErinaInit** [comment]
Now sagelib is good, however sagemath is failing...... Checking for new issues. Closing this one.
